packages/checkout/widgets-lib/src/widgets/connect/views/ReadyToConnect.tsx

```diff
@@ -86,21 +86,27 @@ export function ReadyToConnect({ targetChainId, allowedChains }: ReadyToConnectP
 
   // eslint-disable-next-line @typescript-eslint/no-shadow
   const handleConnectViewUpdate = async (provider: Web3Provider) => {
-    // Skip checks for Passport. Passport will be shipped with the
-    // zkEVM network pre-configured and changes of networks are handled
-    // by the ConnectLoader.
-    // TODO: Remove this check when Passport has support for L1.
-    if (!isPassport) {
-      const chainId = await provider.getSigner().getChainId();
-      if (chainId !== targetChainId && !allowedChains?.includes(chainId)) {
+    const chainId = await provider.provider.request!({ method: 'eth_chainId', params: [] });
+    // eslint-disable-next-line radix
+    const parsedChainId = parseInt(chainId.toString());
+    if (parsedChainId !== targetChainId && !allowedChains?.includes(parsedChainId)) {
+      // TODO: What do we do with Passport here as it can't connect to L1
+      if (isPassport) {
         viewDispatch({
           payload: {
             type: ViewActions.UPDATE_VIEW,
-            view: { type: ConnectWidgetViews.SWITCH_NETWORK },
+            view: { type: ConnectWidgetViews.SUCCESS },
           },
         });
         return;
       }
```

Comment:
@dreamoftrees I was thinking we handle Passport by just removing it from the Wallet list when targetChainId is an L1 chain. This code will currently let it pass through and connect correctly. But we know Passport is only L2 for now.